qml/pages/FirstPage.qml

```diff
@@ -41,67 +70,96 @@ Page {
+
+        delegate: ListItem {
+            id: listItem
+            width: page.width / columnCount
+            property int baseHeight: gridView.cellHeight
+            height: baseHeight + (index === currIndex ? offsetY : 0.0)
+            contentWidth: width
+            contentHeight: height
+
+            Image {
+                id: programThumbnail
+                y: index >= firstOffsetIndex ? offsetY : 0.0
+                z: 1
+                width: parent.width
+                height: width / 16.0 * 9.0
+                sourceSize.width: imageSizeX
+                sourceSize.height: imageSizeY
+                source: modelData.image && modelData.image.id && modelData.image.available
+                        ? "http://images.cdn.yle.fi/image/upload/w_" + imageSizeX + ",h_" + imageSizeY + ",c_fit/" + modelData.image.id + ".jpg"
+                        : ""
+                BusyIndicator {
+                    anchors.centerIn: parent
+                    size: BusyIndicatorSize.Medium
+                    running: parent.status !== Image.Ready && parent.status !== Image.Error
+                    visible: running
+                }
+                Image {
+                    anchors.centerIn: parent
+                    visible: parent.status === Image.Error
+                    source: "image://theme/icon-l-image"
+                }
+            }
+
+            Label {
+                id: programLabel
+                x: Theme.paddingSmall
+                anchors.top: programThumbnail.bottom
+                width: programThumbnail.width
+                truncationMode: TruncationMode.Fade
+                text: modelData.title
+                font.pixelSize: Theme.fontSizeMedium
             }
-            delegate: ListItem {
-                id: listItem
-                contentHeight: column.height + Theme.paddingMedium
-                contentWidth: listView.width
-
-                menu: ContextMenu {
-                    MenuItem {
-                        text: qsTr("Show program info")
-                        onClicked: {
-                            pageStack.push(Qt.resolvedUrl("ProgramOverviewPage.qml"), {
-                                               "program": modelData
-                                           })
-                        }
-                    }
-                    MenuItem {
-                        visible: Boolean(modelData.seriesId)
-                        text: qsTr("Show programs in series")
-
-                        onClicked: pageStack.push(Qt.resolvedUrl("ProgramsPage.qml"), {
-                                                      "series": modelData
-                                                  })
-                    }
+
+            menu: ContextMenu {
+                id: contextMenu
+
+                // Using offsetY the gridView size changes dynamically. Magic!
+                onHeightChanged: offsetY = height
```

Comment:
I think this might be causing an issue I observe. If I long press the first item in list to show context menu and then try to click one of the items in the context menu the context menu offset changes before the click is registered and it will instead click the image and video will start to play instead of showing the new page based on menu click.